### DISCUSSION TITLE
How to specify CPU architecture

### DISCUSSION THREAD

**Discussion by zabdoul60460** [BowlerHatLLC/vscode-as3mxml#628] [Q&A] [answered]
Hi,
I try to found how to compile a native Air application for Mac OS 64 bits CPU (intel, M1, M2).
I found an AIR options apply to Android applications but not for Mac OS.
Could you help me ?
Best regards.

**Comment by joshtynjala** [comment, marked as answer]
[Create a new ActionScript project in Visual Studio Code that targets Adobe AIR for desktop platforms](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/Create-a-new-ActionScript-project-in-Visual-Studio-Code-that-targets-Adobe-AIR-for-desktop-platforms)

**Comment by zabdoul60460** [comment]
I use a config file like below :

```
{
	"config": "air",
	"type": "app",

	
	"compilerOptions": {
		"source-path": [
			"src"
		
		],
		"library-path": [
			"libs",
			"linked-swc"
		],
		
		"output": "bin/MyApp.swf",

		"strict": false
		


	},
	"mainClass": "MyApp",
	"application": "src/MyApp-app.xml",
	"additionalOptions": "-locale=fr_FR,en_US",
	"copySourcePathAssets": true,
	
"airOptions": {
	
	"target": "native",
	"output": "bin/MyApp-Build2208c.dmg",
	"signingOptions": {
		"storetype": "pkcs12",
		"keystore": "SE_key.p12"
		
	
	},

	"files": [
		{"file": "src/app_icons/Logo16.png","path": "app_icons/Logo16.png"},
		{"file": "src/app_icons/Logo32.png","path": "app_icons/Logo32.png"},
		{"file": "src/app_icons/Logo48.png","path": "app_icons/Logo48.png"},
		{"file": "src/installeurs","path": "installeurs"}

	]
	
}
}
```

But with this config file, I need to specify 'Start with rosetta', if not, it's impossible to start air application !

Thanks for helping

**Reply by joshtynjala** [reply]
> need to specify 'Start with rosetta', if not, it's impossible to start air application !

Has HARMAN added support for the Apple Silicon architecture to AIR yet? I don't recall hearing that they had already. However, I don't follow AIR as closely as I used to, so maybe I missed it.

If Apple Silicon is not yet supported by AIR, then this is the expected behavior. To run Intel architecture on M1/M2 CPU, you need to use Rosetta.

**Reply by joshtynjala** [reply]
By the way, I recommend using "bundle" instead of "native" when targeting desktop. It's better to use the captive runtime.

**Reply by zabdoul60460** [reply]
if I use bundle instead of native, may I have only one compilation for both OS (Windows and Mac OS) ?
Best regards

**Reply by joshtynjala** [reply]
No, you would need to create a Windows bundle on Windows and a macOS bundle on macOS.

**Reply by zabdoul60460** [reply]
OK but  native process doesn't work with bundle !
Best regards

**Reply by joshtynjala** [reply]
Are you sure? That doesn't sound right to me.

**Reply by joshtynjala** [reply]
I just checked the AIR documentation, and I confirmed that captive runtime bundles support the extendedDesktop profile required for NativeProcess.
